client/src/__tests__/reducer.test.js

```diff
@@ -0,0 +1,53 @@
+import { createStore } from 'redux';
+import defer from 'lodash/defer';
+import reducers from '../reducer';
+import { newDataReceivedAction, globalMessageAction } from '../action'
+import { ERROR_MESSAGE, WARNING_MESSAGE, NOTICE_MESSAGE } from '../constant'
+
+const store = createStore(reducers);
+
+describe('reducer.js', () => {
+  it('test storeData reducer', async () => {
+    const inventoryDetails = {
+      store: 'My store',
+      model: 'My shoe',
+      inventory: 10,
+    };
+    store.dispatch(newDataReceivedAction(inventoryDetails));
```

Comment:
Reducers should be tested directly, without dispatching events to a Redux store. Your introducing side effects in your unit tests with Redux's state and actions. For example, a bug in an action function will result in failing reducer test. 

Instead, the reducer could be tested like that:
```javascript
const newState = storeData({}, { type: NEW_DATA_RECEIVED, payload: inventoryDetails });
expect(newState).toEqual({
  [inventoryDetails.store]: {
    [inventoryDetails.model]: inventoryDetails.inventory,
  },
});
```